src/host/getset.cpp

```diff
@@ -357,6 +357,17 @@ void ApiRoutines::GetNumberOfConsoleMouseButtonsImpl(ULONG& buttons) noexcept
             WI_ClearFlag(gci.Flags, CONSOLE_USE_PRIVATE_FLAGS);
         }
 
+        if (WI_IsFlagSet(mode, ENABLE_MOUSE_INPUT) && !WI_IsFlagSet(context.InputMode, ENABLE_MOUSE_INPUT))
+        {
+            gci.GetActiveInputBuffer()->PassThroughEnableButtonEventMouseMode(true);
+            gci.GetActiveInputBuffer()->PassThroughEnableSGRExtendedMouseMode(true);
+        }
+        else if (!WI_IsFlagSet(mode, ENABLE_MOUSE_INPUT) && WI_IsFlagSet(context.InputMode, ENABLE_MOUSE_INPUT))
+        {
+            gci.GetActiveInputBuffer()->PassThroughEnableButtonEventMouseMode(false);
+            gci.GetActiveInputBuffer()->PassThroughEnableSGRExtendedMouseMode(false);
+        }
+

```

Comment:
You can probably simplify this pretty well:

```
const auto oldHasMouseMode{ WI_IsFlagSet(context.InputMode, ...) };
const auto newHasMouseMode{ WI_IsFlagSet(mode, ...) };
if (oldHasMouseMode != newHasMouseMode)
{
    PassThroughButtonEvent(newHasMouseMode);
}
```

Reply:
Thanks for the suggestion!